config/samples/gatewayclass.yaml

```diff
@@ -0,0 +1,6 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: GatewayClass
+metadata:
+  name: mctc-gw-istio-instance-per-cluster
```

Comment:
maybe mctc-gw-istio-external?

Reply:
I wonder if 'external' has connotations with the gateway being outside the data plane clusters?
E.g. a service by the cloud provider or a 3rd party.
In our POC, the gateway will be internal to the data plane clusters.

If we extrapolate, the different types would be:

* Internal, instance per cluster
* Internal, 1 dedicated config cluster
* Internal, multi cluster, single region, completely managed and hidden from user
* Internal, multi cluster, multi region, completely managed and hidden from user
* External, provided by 3rd party

Reply:
hmm. I had thought of external as allowing traffic into the cluster from public internet

Reply:
Yes, that lines up better with existing terminology around load balancers & gateways.

What about `mctc-gw-istio-external-instance-per-cluster`

Other types could be:
* `mctc-gw-istio-external-config-cluster`
* `mctc-gw-amazonapigw-external`

Reply:
yeah I am ok with that for now